core/src/main/scala/neotypes/DeferredQuery.scala

```diff
@@ -1,465 +1,211 @@
-    * @tparam F effect type.
-    * @return An effectual value that will execute the query.
-    */
   def execute[F[_]](tx: Transaction[F])
                    (implicit em: ExecutionMapper[T]): F[T] =
     tx.execute(query, params)
 
-  /** Evaluate the query and get the results as a Stream.
-    *
-    * @see <a href="https://neotypes.github.io/neotypes/docs/streams.html">The streaming documentation</a>.
-    *
-    * @param tx neotypes transaction.
-    * @param rm result mapper for type T.
-    * @tparam S stream type.
-    * @tparam F effect type.
-    * @return An effectual Stream of T values.
-    */
+
   def stream[S[_], F[_]](tx: StreamingTransaction[S, F])
                         (implicit rm: ResultMapper[T]): S[T] =
     tx.stream(query, params)
+}
+
+case class DeferredCommand[T](
+     query: String,
+     params: Map[String, QueryParam],
+     paramLocations: List[Int]) extends Deferred[T] {
+  def driverTransactionFC[F[_], C[_]](driver: Driver[F])(tx: Transaction[F] => F[C[T]]): F[C[T]] = {
+    driver.transact(tx)
+  }
+  def driverTransactionConfigFC[F[_], C[_]](driver: Driver[F])(config: TransactionConfig, tx: (Transaction[F] => F[C[T]])): F[C[T]] = {
+    driver.transact(config)(tx)
+  }
+
+  def streamDriverTransaction[S[_], F[_]](driver: StreamingDriver[S, F])(st: StreamingTransaction[S, F] => S[T]): S[T] = {
+    driver.streamingTransact(st)
+  }
+  def streamDriverTransactionConfig[S[_], F[_]](driver: StreamingDriver[S, F])(config: TransactionConfig, st: StreamingTransaction[S, F] => S[T]): S[T] = {
+    driver.streamingTransact(config)(st)
+  }
+
+  def driverTransactionF[F[_]](driver: Driver[F])(tx: Transaction[F] => F[T]): F[T] = driver.transact(tx)
+
+  def driverTransactionConfigF[F[_], C[_]](driver: Driver[F])(config: TransactionConfig, tx: (Transaction[F] => F[T])): F[T] = driver.transact(config)(tx)
 
-  /** Executes the query and returns a custom collection of values.
-    *
-    * @example
-    * {{{
-    * val result: F[ListMap[Person, Movie]] =
-    *   "MATCH (p:Person {name: 'Charlize Theron'})-[]->(m:Movie) RETURN p,m"
-    *     .query[(Person, Movie)]
-    *     .collectAs(ListMap)(driver)
-    * }}}
-    *
-    * @param factory the type of the custom collection to return.
-    * @tparam C collection type to be constructed.
-    * @return An effectual value that will compute a Collection of T elements.
-    */
   def collectAs[C](factory: Factory[T, C]): CollectAsPartiallyApplied[T, C] =
     new CollectAsPartiallyApplied(factory -> this)
 
-  /** Creates a new query with an updated set of parameters.
-   *
-    * @note If params contains a key that is already present in the current query,
-    * the new one will override the previous one.
-    *
-    * @param params QueryParams to be added.
-    * @return DeferredQuery with params added to existing params.
-    */
-  def withParams(params: Map[String, QueryParam]): DeferredQuery[T] =
+  def withParams(params: Map[String, QueryParam]): Deferred[T] =
```

Comment:
I believe you can actually override this and make the return type precise to this specific subclass.